### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert false positive

### ISSUE THREAD

**Issue by blerner** [pmd/pmd#1365]
**Affects PMD Version:** 6.7.0

**Rule:** JUnitTestsShouldIncludeAssert

**Description:** The following two test cases are identical, but for the precise text of the annotation before them.  But they give different warnings in PMD.  My students encountered this with PMD v6.2.0, and I've just confirmed locally that it still occurs on 6.7.0.

**Code Sample demonstrating the issue:**
```java
import org.junit.Test;

class Style {
  // This triggers the JUnitTestsShouldIncludeWarning rule
  @org.junit.Test(expected = IllegalArgumentException.class)
  public void moveOutOfBoundsFrom() {
    doSomething();
  }
  // This does not
  @Test(expected = IllegalArgumentException.class)
  public void moveOutOfBoundsFrom() {
    doSomething();
  }
}
```

**Running PMD through:** Other (custom too invoking `PMD.main` programmaticallyl, referencing `<rule ref="category/java/bestpractices.xml/JUnitTestsShouldIncludeAssert" />` to load this rule)

**Comment by jsotuyod** [comment]
@blerner if I'm reading the code correctly, it should work as long as you provide the auxclasspath (which tools such as Gradle and Maven can do automatically).

The code could be better on side (simpler and better for these scenarios), but the bottom line remains: for PMD to be most accurate, **auxclasspath data has to be provided**.

https://github.com/pmd/pmd/blob/c62e4beec30d65c868fe05842c3fe4d7593cba23/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/JUnitTestsShouldIncludeAssertRule.java#L109-L110

can be simplified to:

```java
if (name != null && TypeHelper.isA(name, JUNIT4_CLASS_NAME)) {
```

which should also work on this scenario.

**Comment by blerner** [comment]
Ah!  I can do that :)  I'm already providing a couple of jars on the auxclasspath (to custom libraries that our courses use); it didn't occur to me to put JUnit's jar on the auxclasspath as well.  Thanks for the quick triage.
